retail/interactive-tutorial/RetailEvents.Samples/ImportUserEventsGcsSample.cs

```diff
@@ -0,0 +1,132 @@
+
+        ImportUserEventsRequest importRequest = new ImportUserEventsRequest
+        {
+            Parent = defaultCatalog,
+            InputConfig = new UserEventInputConfig
+            {
+                GcsSource = gcsSource
+            },
+            ErrorsConfig = new ImportErrorsConfig
+            {
+                GcsPrefix = gcsErrorsBucket
+            }
+        };
+
+        Console.WriteLine("Import user events from google cloud source. request:");
+        Console.WriteLine(importRequest);
+        Console.WriteLine();
+
+        return importRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to import user events.
+    /// </summary>
+    /// <param name="projectId">The current project id.</param>
+    public Operation<ImportUserEventsResponse, ImportMetadata> ImportUserEventsFromGcs(string projectId, string bucketName = null)
+    {
+        ImportUserEventsRequest importGcsRequest = GetImportUserEventsGcsRequest(GcsEventsObject, projectId, bucketName);
+        UserEventServiceClient client = UserEventServiceClient.Create();
+        Operation<ImportUserEventsResponse, ImportMetadata> importResponse = client.ImportUserEvents(importGcsRequest);
+
+        Console.WriteLine("The operation was started:");
+        Console.WriteLine(importResponse.Name);
+        Console.WriteLine();
+
+        Console.WriteLine("Please wait till opeartion is done");
+        Console.WriteLine();
+
+        Operation<ImportUserEventsResponse, ImportMetadata> importResult = importResponse.PollUntilCompleted();
+
+        Console.WriteLine("Import user events operation is done");
+        Console.WriteLine();
+
+        Console.WriteLine("Number of successfully imported events: " + importResult.Metadata.SuccessCount);
+        Console.WriteLine("Number of failures during the importing: " + importResult.Metadata.FailureCount);
+        Console.WriteLine();
+        Console.WriteLine("Operation result:");
+        Console.WriteLine(importResult.Result);
+        Console.WriteLine();
+
+        return importResult;
+    }
+}
+// [END retail_import_user_events_from_gcs]
+
+/// <summary>
+/// The import user events gcs tutorial class.
+/// </summary>
+public static class ImportUserEventsGcsTutorial
+{
+    [Runner.Attributes.Example]
+    public static Operation<ImportUserEventsResponse, ImportMetadata> ImportUserEventsFromGcs()
+    {
+        string projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        var sample = new ImportUserEventsGcsSample();
+
+        string createdBucketName = null;
+
+        EventsDeleteGcsBucket.PerformDeletionOfEventsBucketName(createdBucketName);
+        createdBucketName = EventsCreateGcsBucket.PerformCreationOfEventsGcsBucket();
+
+        Operation<ImportUserEventsResponse, ImportMetadata> importResponse = sample.ImportUserEventsFromGcs(projectId, createdBucketName);
+
+        EventsDeleteGcsBucket.PerformDeletionOfEventsBucketName(createdBucketName);
```

Comment:
You can't do this.
The bucket being used is the same for all test runs, and test runs may happen at the same time for many reasons, so test run A may delete the bucket just after test run B has created it, which we'll make test run B fail.

All the othe retail samples and tests have assumed that the data is already there (catalogs, etc.), so these tests should do the same. Assume the bucket is there and assume it contains the data you need, and then write your samples and tests. Also, that's the reason why we can separate out the samples + tests, from the setup and teardown.